src/main/java/newitem/NewItemController.java

```diff
@@ -0,0 +1,17 @@
+package newitem;
+
+public class NewItemController {
+
+    private final NewItemInteractor newItemInteractor;
+
+    public NewItemController() {
+        this.newItemInteractor = new NewItemInteractor();
+    }
+
+    //returns the item UPC mostly for testing purposes but also maybe that's what the user will be given upon
+    //execution of the request
+    public long newItem(String name, long UPC, int price) {
```

Comment:
the way i did it was i had one method for lookup by upc, and one for lookup by name

i also think it might be better to return the full Product and not just the upc, that way the presenter can display all of the information about the product more easily